src/masternodes/proposals.h

```diff
@@ -0,0 +1,151 @@
+std::string CPropVoteToString(const CPropVoteType status);
+std::string CPropStatusToString(const CPropStatusType status);
+
+struct CCreatePropMessage {
+    CPropType type;
+    CScript address;
+    CAmount nAmount;
+    uint8_t nCycles;
+    std::string title;
+    int32_t blocksCount;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        uint8_t typeByte;
+        if (ser_action.ForRead()) {
+            READWRITE(typeByte);
+            type = static_cast<CPropType>(typeByte);
+        } else {
+            typeByte = static_cast<uint8_t>(type);
+            READWRITE(typeByte);
+        }
+        READWRITE(address);
+        READWRITE(nAmount);
+        READWRITE(nCycles);
+        READWRITE(title);
+        READWRITE(blocksCount);
+    }
+};
+
+struct CPropVoteMessage {
+    CPropId propId;
+    uint256 masternodeId;
+    CPropVoteType vote;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(propId);
+        READWRITE(masternodeId);
+        uint8_t voteByte;
+        if (ser_action.ForRead()) {
+            READWRITE(voteByte);
+            vote = static_cast<CPropVoteType>(voteByte);
+        } else {
+            voteByte = static_cast<uint8_t>(vote);
+            READWRITE(voteByte);
+        }
+    }
+};
+
+struct CPropObject : public CCreatePropMessage {
+    // memory only
+    uint8_t cycle;
+    uint32_t creationHeight;
+    uint32_t finalHeight;
+    // memory only
+    CPropStatusType status;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITEAS(CCreatePropMessage, *this);
+        READWRITE(creationHeight);
+        READWRITE(finalHeight);
+    }
+};
+
+struct CMnVotePerCycle {
+    CPropId propId;
+    uint8_t cycle;
+    uint256 masternodeId;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(propId);
+        READWRITE(cycle);
+        READWRITE(masternodeId);
+    }
+};
+
+/// View for managing proposals and their data
+class CPropsView : public virtual CStorageView
+{
+public:
+    ~CPropsView() override = default;
+
+    Res CreateProp(const CPropId& propId, uint32_t height, const CCreatePropMessage& prop);
+    Optional<CPropObject> GetProp(const CPropId& propId);
+    Res UpdatePropCycle(const CPropId& propId, uint8_t cycle);
+    Res UpdatePropStatus(const CPropId& propId, uint32_t height, CPropStatusType status);
+    Res AddPropVote(const CPropId& propId, const uint256& masternodeId, CPropVoteType vote);
+    Optional<CPropVoteType> GetPropVote(const CPropId& propId, uint8_t cycle, const uint256& masternodeId);
```

Comment:
`GetPropVote` is currently unused.

Reply:
Yes, it's still unused. Probably in the future.